app/src/main/java/org/standardnotes/notes/MainActivity.kt

```diff
@@ -107,6 +132,39 @@ class MainActivity : BaseActivity(), SyncManager.SyncListener {
         selectedTagId = selectedId // In case selected tag wasn't found in list
     }
 
+    fun removeDrawerToggle() {
+        drawerToggle.isDrawerIndicatorEnabled = false
+        supportActionBar?.setDisplayHomeAsUpEnabled(false)
+        supportActionBar?.setHomeButtonEnabled(false)
+    }
+
+    override fun addDrawerToggle() {
+        drawerToggle.isDrawerIndicatorEnabled = true
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.setHomeButtonEnabled(true)
+    }
+
+    override fun newNoteListener(uuid: String) {
+        noteFragment = NoteFragment()
+        noteFragment!!.detachListener = this
+        val bundle = Bundle()
+        bundle.putString(NoteListFragment.EXTRA_NOTE_ID, uuid)
+        noteFragment!!.arguments = bundle
+        if (findViewById(R.id.note_container) == null) {
+            removeDrawerToggle()
```

Comment:
Possible solution - push the setup into NoteListFragment.setupActionBar(ActionBar, Drawer) and NoteFragment.setupActionBar(ActionBar, Drawer) methods, and have them do what is currently in those *DrawerToggle() methods. Then whenever the fragment in note_list_container changes, call its setupAcitonBar() and have it do whatever changes needed. 

note_list_container probably needs renaming to primary_container or something like that.

You also probably want to keep the back arrow when it's in the single_pane mode and showing a note, and have it do the right thing. Possibly those fragments need to have a onBackPressed() method too.